### PR TITLE
Migrate SVM Predict to DPC++ interfaces

### PR DESCRIPTION
**Pull request by dmitrii-kriukov**

### CODE REVIEW

File: cpp/oneapi/dal/algo/rbf_kernel/backend/gpu/compute_kernel_dense_dpc.cpp [hunk 1]
```diff
@@ -72,48 +74,55 @@ inline auto compute_exponents(sycl::queue& queue,
 }
 
 template <typename Float>
-inline auto compute_rbf(sycl::queue& queue,
-                        const pr::ndview<Float, 2>& x_nd,
-                        const pr::ndview<Float, 2>& y_nd,
-                        pr::ndview<Float, 2>& res_nd,
-                        double sigma,
-                        const dal::backend::event_vector& deps = {}) {
+void compute_rbf(sycl::queue& queue,
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/rbf_kernel/backend/gpu/compute_kernel_dense_dpc.cpp:77]: Maybe it makes sense to return sycl::event?

File: cpp/oneapi/dal/algo/rbf_kernel/detail/compute_ops.hpp [hunk 2]
```diff
@@ -19,6 +19,10 @@
 #include "oneapi/dal/algo/rbf_kernel/compute_types.hpp"
 #include "oneapi/dal/detail/error_messages.hpp"
 
+#ifdef ONEDAL_DATA_PARALLEL
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/rbf_kernel/detail/compute_ops.hpp:22]: ???

File: cpp/oneapi/dal/algo/sigmoid_kernel/backend/cpu/compute_kernel.hpp [hunk 3]
```diff
@@ -19,13 +19,25 @@
 #include "oneapi/dal/algo/sigmoid_kernel/compute_types.hpp"
 #include "oneapi/dal/backend/dispatcher.hpp"
 
+#ifdef ONEDAL_DATA_PARALLEL
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/sigmoid_kernel/backend/cpu/compute_kernel.hpp:22]: ???

File: cpp/oneapi/dal/algo/sigmoid_kernel/backend/gpu/compute_kernel.hpp [hunk 4]
```diff
@@ -19,13 +19,25 @@
 #include "oneapi/dal/algo/sigmoid_kernel/compute_types.hpp"
 #include "oneapi/dal/backend/dispatcher.hpp"
 
+#ifdef ONEDAL_DATA_PARALLEL
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/sigmoid_kernel/backend/gpu/compute_kernel.hpp:22]: ???

File: cpp/oneapi/dal/algo/sigmoid_kernel/detail/compute_ops.hpp [hunk 5]
```diff
@@ -19,6 +19,10 @@
 #include "oneapi/dal/algo/sigmoid_kernel/compute_types.hpp"
 #include "oneapi/dal/detail/error_messages.hpp"
 
+#ifdef ONEDAL_DATA_PARALLEL
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/sigmoid_kernel/detail/compute_ops.hpp:22]: ???

File: cpp/oneapi/dal/algo/svm/backend/gpu/infer_cls_kernel_dpc.cpp [hunk 6]
```diff
@@ -33,83 +30,122 @@ using input_t = infer_input<task::classification>;
 using result_t = infer_result<task::classification>;
 using descriptor_t = detail::descriptor_base<task::classification>;
 
-namespace daal_svm = daal::algorithms::svm;
-namespace daal_kernel_function = daal::algorithms::kernel_function;
-namespace interop = dal::backend::interop;
+namespace pr = dal::backend::primitives;
+namespace be = dal::backend;
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/svm/backend/gpu/infer_cls_kernel_dpc.cpp:34]: namespace bk?

File: cpp/oneapi/dal/algo/svm/backend/gpu/infer_cls_kernel_dpc.cpp [hunk 7]
```diff
@@ -33,83 +30,122 @@ using input_t = infer_input<task::classification>;
...
 
 template <typename Float>
-using daal_svm_predict_kernel_t =
-    daal_svm::prediction::internal::SVMPredictImplOneAPI<daal_svm::prediction::defaultDense, Float>;
+auto make_responses(sycl::queue& q,
+                    const pr::ndarray<Float, 1>& distances,
+                    const int64_t first_class_response,
+                    const int64_t second_class_response,
+                    const be::event_vector& deps = {}) {
+    const auto size = distances.get_count();
```
Comment (amgrigoriev) [inline cpp/oneapi/dal/algo/svm/backend/gpu/infer_cls_kernel_dpc.cpp:42]: Dimension check for distances?

**Review by amgrigoriev — APPROVED** [review]

**Review by vlad-nazarov — APPROVED** [review]

**Pull request merged**